mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -57,6 +57,18 @@ static TosaOp createOpAndInfer(mlir::PatternRewriter &rewriter,
   return op;
 }
 
+static tosa::CastOp createCastOp(PatternRewriter &rewriter, Location loc,
+                                 Type resElementType, Value input) {
+  ShapedType inputType = input.getType().cast<ShapedType>();
+  auto elementType = inputType.getElementType();
+  Type resType = inputType.cloneWith({}, resElementType);
+
+  auto op = rewriter.create<tosa::CastOp>(loc, elementType, input);
```

Comment:
Is there a form of this builder that takes the result type? The fact that you're calling `setType()` feels off.

Reply:
(but maybe that's just how Tosa stuff goes)

Reply:
You caught a bug. I was trying to match with the generate op creation function in a few lines above. It is only after this comment that I realize the CastOp's creation needs to take the result type, not input element type.

Reply:
Done in 8aed021